### PR TITLE
CLI: implement authenticated calls to MI Management API

### PR DESCRIPTION
**Pull request by menuka94**
## Usage (Adding a new MI instance and logging in to it)
First, add a new remote.
```
mi remote add TestServer1 localhost 9164
```

Then, select `TestServer1` as the current remote (`remote login` only works for the current remote).

```
mi remote select TestServer1
```

Now you can log in to the remote `TestServer1` using the newly introduced`login` command.
Login:
```
mi remote login # will be prompted for username and password
```

Login (with in-line username and password):
```
mi remote login admin admin
```

Logout:
```
mi remote logout
```

### CODE REVIEW

**Comment by Nirothipan** [comment]
@menuka94 how are we handling multiple server scenarios here

**Comment by menuka94** [comment]
> @menuka94 how are we handling multiple server scenarios here

'remote login` command works only for the current remote. If the user wishes to login to a different remote, first, the current remote should be changed to it.

File: cmd/README.md [hunk 1]
```diff
@@ -62,11 +62,13 @@ Command Line tool for managing the WSO2 Micro Integrator
...
-            remove [nick-name]                   Remove a Micro Integrator
-            update [nick-name] [host] [port]     Update a Micro Integrator
-            select [nick-name]                   Select a Micro Integrator on which commands are executed
-            show                                 Show available Micro Integrators
+            add [name] [host] [port]        Add a Micro Integrator
+            remove [name]                   Remove a Micro Integrator
+            update [name] [host] [port]     Update a Micro Integrator
+            select [name]                   Select a Micro Integrator instance on which commands are to be executed
+            show                            Show added Micro Integrator instances
+            login                           Login to the current Micro Integrator instance
```
Comment (sdkottegoda) [inline cmd/README.md]: login [username] [password] ?

Comment (menuka94) [reply cmd/README.md]: Updated.

**Comment by menuka94** [comment]
Please note that https://github.com/wso2/micro-integrator/pull/842 has to be merged for the `login` functionality to work.

**Pull request merged**